### DISCUSSION TITLE
`github.rest.issues.createComment` results in "Resource not accessible by integration" error

### DISCUSSION THREAD

**Discussion by jprider63** [actions/github-script#490] [Q&A] [answered]
I'm [attempting](https://github.com/rems-project/cerberus/pull/537) to use `github-script` to [post a comment](https://github.com/rems-project/cerberus/pull/537/files#diff-012caac8e221f6dbc3f5446e6b2e33d60dd7568b8bda9394cf2d6e5355652a31R150) to a PR during CI. Unfortunately it is [failing](https://github.com/rems-project/cerberus/actions/runs/11035157875/job/30650446031) with the error `"Error: Unhandled error: HttpError: Resource not accessible by integration"` when run on a PR to the main repository. When the same commit (`72c2497`) is run on my fork of the repository, [CI passes and posts the comment](https://github.com/GaloisInc/cerberus/pull/5#issuecomment-2374273441) without error. So far, I've tried adding the following permission settings, but none of them work for PR's against the main repository: 

```
permissions:
  pull-requests: write
```

```
permissions:
  pull-requests: write
  issues: write
```

```
permissions: write-all
```

```
permissions:
  id-token: write
  contents: read
  issues: write
  pull-requests: write
```

We've also checked that "Allow all actions and reusable workflows" and "Read and write permissions" is set for the main repository's action permission settings. 

Is there another permission that we need to set? Any suggestions would be appreciated! Thanks!

**Comment by HowieHz** [comment]
Use of [information passed between jobs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/passing-information-between-jobs)

An example of problem solving https://github.com/travellings-link/travellings/commit/6fc44182f6b49bac4105b8ff9687ac157508ea4d (Ignore the job named “comment” and focus on "comment-auto-check-assistant-result" and "auto-check-assistant".)

**Comment by joshmgross** [comment, marked as answer]
Pull Requests workflows from forks can't utilize write permissions, which is why you're getting a failure. You can see the effective permissions for a job in the "Set up job" step logs.

[`pull_request_target`](https://docs.github.com/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target) is likely the event you want, but please read through the full documentation to understand the security risks. If you have any questions about that, feel free to ask!
